### PR TITLE
Add using for join builder, update some comments

### LINKED ISSUE #633: Feature: Add support for USING with join
I see it in arg types, but I'm not sure how to leverage it.

```
 sg.select(
    PARTITION.quote(),
    PATIENT_ID,
    OR_CASE_ID
).from_('merged_df').join('vte_diagnosis_df', using='patient_id, encounter_id')
```

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #633

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -1641,6 +1683,11 @@ def join(
             on = and_(*ensure_list(on), dialect=dialect, **opts)
             join.set("on", on)
 
+        if using:
+            if isinstance(using, str):
+                using = [to_identifier(name) for name in using.split(", ")]
```
Comment (barakalon) [inline sqlglot/expressions.py:1688]: Why are we accepting a CSV instead of a list of strings?

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -937,6 +937,42 @@ def on(self, *expressions, append=True, dialect=None, copy=True, **opts):
 
         return join
 
+    def using(self, *expressions, append=True, dialect=None, copy=True, **opts):
```
Comment (barakalon) [inline sqlglot/expressions.py:940]: This isn't tested.

Comment (GeorgeSittas) [reply sqlglot/expressions.py:940]: Thanks for pointing out, I forgot to add more tests after the docstring.

File: sqlglot/expressions.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlglot/expressions.py:1688]: Right, I'll fix this too. Didn't notice that's what we were doing for `on` due to docstring type signature confusion. Thanks!

Comment (tobymao) [reply sqlglot/expressions.py:1688]: you can call ensure_list

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -1641,6 +1683,11 @@ def join(
             on = and_(*ensure_list(on), dialect=dialect, **opts)
             join.set("on", on)
 
+        if using:
+            if isinstance(using, str):
```
Comment (tobymao) [inline sqlglot/expressions.py:1687]: what if it's a list of strings?, ensure list should be first

Comment (GeorgeSittas) [reply sqlglot/expressions.py:1687]: Can we also accept a list of expressions? Is that a case we should consider?

Comment (barakalon) [reply sqlglot/expressions.py:1687]: I guess someone could pass an Identifier or a list of Identifiers, so yeah?

**Review by tobymao — APPROVED** [review]

**Pull request merged**